core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -423,7 +445,13 @@ Builder scanMetrics(ScanMetrics newScanMetrics) {
       return this;
     }
 
-    DeleteFileIndex build() {
+    private Collection<DeleteFile> filterDeleteFiles() {
+      return Streams.stream(deleteFiles)
+          .filter(file -> file.dataSequenceNumber() > minSequenceNumber)
+          .collect(Collectors.toList());
```

Comment:
Question: does it make sense to keep this iterable instead of materializing to a List?  (looks like we just do a for: each here)

Reply:
Need a collection later to call size on it.

Reply:
We could replace it later but it would be a separate PR to adapt how scan metrics are being computed.